### PR TITLE
Post-release fixes

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/MaKaC/PDFinterface/conference.py [hunk 1]
```diff
@@ -410,8 +410,20 @@ class ContributionBook(PDFLaTeXBase):
...
+        if sort_by == 'schedule':
+            key_func = lambda c: (c.start_dt is None, c.start_dt)
+        elif sort_by == 'sessionTitle':
+            key_func = lambda c: (c.session is None, c.session.title.lower() if c.session else '')
+        elif sort_by == 'speaker':
+            def key_func(c):
+                speakers = c.speakers
+                if not c.speakers:
+                    return True, None
+                return False, speakers[0].get_full_name(last_name_first=True, last_name_upper=False,
```
Comment (nop33) [inline indico/MaKaC/PDFinterface/conference.py]: `last_name_first=True` is already the default

File: indico/MaKaC/webinterface/tpls/latex/book_of_abstracts.tpl [hunk 2]
```diff
@@ -21,10 +21,10 @@
 <%block name="book_body">
     % for contrib in contribs:
         % if contrib.can_access(_session.user):
-            \fancyhead[L]{\small \rmfamily \color{gray} \truncateellipses{${conf.getTitle() | latex_escape}}{300pt} / ${_("Book of Abstracts")}}
+            \fancyhead[L]{\small \rmfamily \color{gray} \truncateellipses{${conf.title | latex_escape}}{300pt} / ${_("Book of Abstracts")}}
             \fancyhead[R]{}
 
-            \addcontentsline{toc}{chapter}{${contrib.title | latex_escape} ${('{0}').format(contrib.getId()) if conf.getBOAConfig().getShowIds() else ''}
+            \addcontentsline{toc}{chapter}{${contrib.title | latex_escape} ${('{0}').format(contrib.id) if conf.getBOAConfig().getShowIds() else ''}
```
Comment (nop33) [inline indico/MaKaC/webinterface/tpls/latex/book_of_abstracts.tpl]: Why not just `unicode(contrib.id)`? The `.format` seems a bit unnecessary since there is no concatenation.

Comment (nop33) [inline indico/MaKaC/webinterface/tpls/latex/book_of_abstracts.tpl]: Ah, you just changed a few characters, you didn't write the line, nvm!

Comment (mic4ael) [inline indico/MaKaC/webinterface/tpls/latex/book_of_abstracts.tpl]: That is true, if we were to change all the occurrences of `'{0}'.format` it would take lots of time since you can find it in many places in the legacy code

File: indico/modules/events/api.py [hunk 3]
```diff
@@ -596,7 +597,7 @@ def contribution(self, idlist):
 
         def _iterate_objs(objIds):
             for objId in objIds:
-                obj = event.contributions.filter_by(id=objId).one()
+                obj = Contribution.query.with_parent(event).filter_by(id=objId).one()
```
Comment (nop33) [inline indico/modules/events/api.py]: Just out of curiosity and interest; What is the reason behind this change?

Comment (ThiefMaster) [inline indico/modules/events/api.py]: event.contributions is not a dynamci relationship anymore

File: indico/modules/events/contributions/util.py [hunk 4]
```diff
@@ -231,6 +231,8 @@ def contribution_type_row(contrib_type):
 @memoize_request
 def get_contributions_with_user_as_submitter(event, user):
     """Get a list of contributions in which the `user` has submission rights"""
+    if not user:
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/util.py]: why not change the code calling it? this function is not supposed to be called without a user

File: indico/modules/events/api.py (hunk 3 above)
Comment (mic4ael) [inline indico/modules/events/api.py]:
```
event_new = db.relationship(
        'Event',
        lazy=True,
        backref=db.backref(
            'contributions',
            primaryjoin='(Contribution.event_id == Event.id) & ~Contribution.is_deleted',
            cascade='all, delete-orphan',
            lazy=True
        )
    )
```
